src/Maths/Silk.NET.Maths/Simd/Simd128.Abs.cs

```diff
@@ -0,0 +1,168 @@
+                
+        
+                return SByte(vector);
+            }            
+            [MethodImpl(Scalar.MaxOpt)]
+            static Vector128<T> SByte(Vector128<T> vector)
+            {
+                if (typeof(T) == typeof(sbyte))
+                {
+#if AdvSIMD
+                    return AdvSimd.Abs(vector.AsSByte()).As<byte, T>();
+#endif
+#if SSE
+                    return Ssse3.Abs(vector.AsSByte()).As<byte, T>();
+#endif
+                }
+                
+        
+                return UInt16(vector);
+            }            
+            [MethodImpl(Scalar.MaxOpt)]
+            static Vector128<T> UInt16(Vector128<T> vector)
+            {
+                if (typeof(T) == typeof(ushort))
+                {
+                    return vector;
+                }
+                
+        
+                return Int16(vector);
+            }            
+            [MethodImpl(Scalar.MaxOpt)]
+            static Vector128<T> Int16(Vector128<T> vector)
+            {
+                if (typeof(T) == typeof(short))
+                {
+#if AdvSIMD
+                    return AdvSimd.Abs(vector.AsInt16()).As<ushort, T>();
+#endif
+#if SSE
+                    return Ssse3.Abs(vector.AsInt16()).As<ushort, T>();
+#endif
+                }
+                
+        
+                return UInt32(vector);
+            }            
+            [MethodImpl(Scalar.MaxOpt)]
+            static Vector128<T> UInt32(Vector128<T> vector)
+            {
+                if (typeof(T) == typeof(uint))
+                {
+                    return vector;
+                }
+                
+        
+                return Int32(vector);
+            }            
+            [MethodImpl(Scalar.MaxOpt)]
+            static Vector128<T> Int32(Vector128<T> vector)
+            {
+                if (typeof(T) == typeof(int))
+                {
+#if AdvSIMD
+                    return AdvSimd.Abs(vector.AsInt32()).As<uint, T>();
+#endif
+#if SSE
+                    return Ssse3.Abs(vector.AsInt32()).As<uint, T>();
+#endif
+                }
+                
+        
+                return UInt64(vector);
+            }            
+            [MethodImpl(Scalar.MaxOpt)]
+            static Vector128<T> UInt64(Vector128<T> vector)
+            {
+                if (typeof(T) == typeof(ulong))
+                {
+                    return vector;
+                }
+                
+        
+                return Single(vector);
+            }            
+            [MethodImpl(Scalar.MaxOpt)]
+            static Vector128<T> Single(Vector128<T> vector)
+            {
+                if (typeof(T) == typeof(float))
+                {
+#if AdvSIMD
+                    return AdvSimd.And(vector.AsSingle(), Vector128.Create(0x7FFF_FFFFu).AsSingle()).As<float, T>();
+#endif
+#if SSE
+                    return Ssse3.And(vector.AsSingle(), Vector128.Create(0x7FFF_FFFFu).AsSingle()).As<float, T>();
+#endif
```

Comment:
I think just calling down to Simd128.And is preferable here?

Reply:
But it doesn't exist 😢 

Hmm

Reply:
I'm fine mering it if it doesn't compile 😛

Reply:
Uh. That would make your job of reviewer much more difficult

Reply:
I can probably add `And` as part of this PR too

Reply:
Actually, as for consistency, wouldn't it make sense to leave it as it is? @HurricanKai . I can do `And` too, but that will require much more time as I will need to PR `Add` for Scalar before

Reply:
(`And` is added)